src/main/java/net/fabricmc/loom/util/Checksum.java

```diff
@@ -60,4 +61,13 @@ public static boolean equals(File file, String checksum) {
 			throw new RuntimeException("Failed to get file hash");
 		}
 	}
+
+	public static String murmur3(File file) {
+		try {
+			HashCode hash = Files.asByteSource(file).hash(Hashing.murmur3_32());
+			return hash.toString();
+		} catch (IOException e) {
+			throw new UncheckedIOException("Failed to get file hash of " + file, e);
+		}
+	}
```

Comment:
Whats the reasoning behind murmur? I would just use something more common md5, sha1 and substring it, chances of a collison would be rare enough.

Reply:
I got it from [a StackOverflow answer](https://stackoverflow.com/a/38678199), and it was already available in Guava and didn't any substringing. I can change it to a more common hash, though, if you want to.

Reply:
Switched it to substringing a SHA256 hash string to 16 characters.

Reply:
Brilliant, that should be way more than enough 👍